currency.sol

```diff
@@ -256,134 +281,176 @@ contract cccur is owned {
     // Prevent changing the Type of a super Admin (2) account
     if (accountType[_targetAccount] != 2){
         limitDebit[_targetAccount] = _debitLimit;
-        limitCredit[_targetAccount] = _creditLimit;
-        
+        limitCredit[_targetAccount] = _creditLimit;   
     }
+
     accountType[_targetAccount] = _accountType;
+    use(_targetAccount);
     
     emit SetAccountParams(now, _targetAccount, _accountStatus, accountType[_targetAccount], limitDebit[_targetAccount],  limitCredit[_targetAccount]);
     
     // ensure the ETH level of the account
+    refill();
     topUp(_targetAccount);
-    topUp(msg.sender);
   }
+
+  function AllowReplaceBy(address target) public payable {
+     if (!actif) revert();                                                      // panic lock
+     if (newAddress[msg.sender]!=address(0)) revert();                          // Already replaced account cannot be replaced again
+     if (!accountStatus[msg.sender]) revert();                                  // locked account cannot be replaced
+     if (accountAlreadyUsed[target]==true) revert();                            // only new account can be a replacement target
+     if (ReplacementRequestNumber[msg.sender]>2) revert();                      // limit the number of replacement request possible
+
+     if (requestReplacementTo[target]!=address(0)) {                            // Cancel replacement request if exists
+       requestReplacementFrom[requestReplacementTo[target]]=address(0);
+       requestReplacementTo[target]=address(0);
+     }
+
+     if (requestReplacementFrom[msg.sender]!=address(0)) {                      // Cancel replacement request if exists
+       requestReplacementTo[requestReplacementFrom[msg.sender]]=address(0);
+       requestReplacementFrom[target]=address(0);
+     }
+
+     requestReplacementFrom[target]=msg.sender;                                // register the request   
+     requestReplacementTo[msg.sender]=target;
+     
+     ReplacementRequestNumber[msg.sender]+=1;
+     topUp(target);                                                            // ensure targuet has eth to accept the request
+  } 
+
+  function CancelReplaceBy() public  {
+     if (!actif) revert();                                                      // panic lock
+     if (requestReplacementFrom[msg.sender]!=address(0)) {                      // Cancel replacement request if exists
+       requestReplacementTo[requestReplacementFrom[msg.sender]]=address(0);
+       requestReplacementFrom[msg.sender]=address(0);
+       refill();  
+     }
+
+  }
+  
   
   /* replace the current account by a new one transfering its content */
   
-  function replaceAccount(address _replacementAccount) public {
+  function AcceptReplaceAccount() public {
```

Comment:
This is not safe for me. I was thinking that the other side had to state from which address it was replacing the account. Indeed, in this current implementation (with 2 tables) : anyone can substitute it's own account before the replacement is actually done. Without a clear re-statement of the replacement account here it seems to me to be a big security issue.